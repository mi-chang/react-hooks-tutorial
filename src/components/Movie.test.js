import React from 'react';
import { render } from '@testing-library/react';
import Movie from './Movie';

describe('Movie', () => {
    it('has default image', () => {
        const { container, getByText, getByAltText, getByRole } = render(<Movie movie={
            {
                Poster: 'N/A',
                Title: 'Default Image Title',
                Year: 1111
            }} />);
        expect(getByText('Default Image Title')).toBeInTheDocument();
        expect(getByText('(1111)')).toBeInTheDocument();
        expect(getByAltText('The movie titled: Default Image Title')).toBeInTheDocument();
        expect(getByRole('img').src).toBe('https://m.media-amazon.com/images/M/MV5BMTczNTI2ODUwOF5BMl5BanBnXkFtZTcwMTU0NTIzMw@@._V1_SX300.jpg');
        expect(container).toMatchSnapshot();
    });

    it('has an image', () => {
        const { container, getByText, getByAltText, getByRole } = render(<Movie movie={
            {
                Poster: 'https://some.com/uri/to/poster/image',
                Title: 'An Image Title',
                Year: 2222
            }} />);
        expect(getByText('An Image Title')).toBeInTheDocument();
        expect(getByText('(2222)')).toBeInTheDocument();
        expect(getByAltText('The movie titled: An Image Title')).toBeInTheDocument();
        expect(getByRole('img').src).toBe('https://some.com/uri/to/poster/image');
        expect(container).toMatchSnapshot();
    });
});
