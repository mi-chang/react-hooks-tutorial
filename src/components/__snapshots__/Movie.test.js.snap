// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie has an image 1`] = `
<div>
  <div
    class="movie"
  >
    <h2>
      An Image Title
    </h2>
    <div>
      <img
        alt="The movie titled: An Image Title"
        src="https://some.com/uri/to/poster/image"
        width="200"
      />
    </div>
    <p>
      (
      2222
      )
    </p>
  </div>
</div>
`;

exports[`Movie has default image 1`] = `
<div>
  <div
    class="movie"
  >
    <h2>
      Default Image Title
    </h2>
    <div>
      <img
        alt="The movie titled: Default Image Title"
        src="https://m.media-amazon.com/images/M/MV5BMTczNTI2ODUwOF5BMl5BanBnXkFtZTcwMTU0NTIzMw@@._V1_SX300.jpg"
        width="200"
      />
    </div>
    <p>
      (
      1111
      )
    </p>
  </div>
</div>
`;
